docs/src/content/docs/concepts/context.mdx

```diff
@@ -5,10 +5,61 @@ sidebar:
   order: 4
 ---
 
-GenSX is a simple framework for building complex generative AI and LLM workflows.
+Contexts provide a way to share data across components without explicitly passing them through props. They're similar to React's Context API but adapted for GenSX workflows.
 
-It uses `jsx` to define and orchestrate workflows with functional, reusable components:
+Contexts are particularly useful for:
+
+- Sharing configuration across multiple components
+- Providing dependencies to deeply nested components without prop drilling
+- Managing state that needs to be accessed by multiple components
+
+GenSX also supports [Providers](../providers) which are a specialized use of contexts that focus on managing configuration and dependencies for your workflow. If you're managing external service configurations like API keys or client instances, you should use providers instead of raw contexts.
```

Comment:
This gives the impression that Providers aren't tightly coupled with context, when they really are. You can't (usefully) use a context without a provider.

Reply:
Very good point. I'll revise this. I probably also need to clean up some of the verbiage in the providers doc